cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -482,53 +486,103 @@ func copyNetworkConfig(src, dst string) error {
+func copyUbuntuDataMisc(srcRoot, dstRoot string) error {
 	for _, globEx := range []string{
 		// systemd's timesync clock file so that the time in recover mode moves
 		// forward to what it was in run mode
 		// NOTE: we don't sync back the time movement from recover mode to run
 		// mode currently, unclear how/when we could do this, but recover mode
 		// isn't meant to be long lasting and as such it's probably not a big
 		// problem to "lose" the time spent in recover mode
-		"system-data/var/lib/systemd/timesync/clock",
+		"var/lib/systemd/timesync/clock",
 	} {
-		if err := copyFromGlobHelper(src, dst, globEx); err != nil {
+		if err := copyFromGlobHelper(srcRoot, dstRoot, globEx); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-// copyUbuntuDataAuth copies the authentication files like
+// copyCoreUbuntuAuthData copies the authentication files like
 //   - extrausers passwd,shadow etc
 //   - sshd host configuration
 //   - user .ssh dir
 //
 // to the target directory. This is used to copy the authentication
 // data from a real uc20 ubuntu-data partition into a ephemeral one.
-func copyUbuntuDataAuth(src, dst string) error {
+func copyCoreUbuntuAuthData(srcUbuntuData, destUbuntuData string) error {
 	for _, globEx := range []string{
 		"system-data/var/lib/extrausers/*",
 		"system-data/etc/ssh/*",
+		// so that users have proper perms, i.e. console-conf added users are
+		// sudoers
+		"system-data/etc/sudoers.d/*",
 		"user-data/*/.ssh/*",
 		// this ensures we get proper authentication to snapd from "snap"
 		// commands in recover mode
 		"user-data/*/.snap/auth.json",
 		// this ensures we also get non-ssh enabled accounts copied
 		"user-data/*/.profile",
-		// so that users have proper perms, i.e. console-conf added users are
-		// sudoers
-		"system-data/etc/sudoers.d/*",
 	} {
-		if err := copyFromGlobHelper(src, dst, globEx); err != nil {
+		if err := copyFromGlobHelper(srcUbuntuData, destUbuntuData, globEx); err != nil {
+			return err
+		}
+	}
+
+	// ensure the user state is transferred as well
+	srcState := filepath.Join(srcUbuntuData, "system-data/var/lib/snapd/state.json")
+	dstState := filepath.Join(destUbuntuData, "system-data/var/lib/snapd/state.json")
+	err := state.CopyState(srcState, dstState, []string{"auth.users", "auth.macaroon-key", "auth.last-id"})
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return fmt.Errorf("cannot copy user state: %v", err)
+	}
+
+	return nil
+}
+
+// copyHybridUbuntuDataAuth copies the authentication files that are relevant on
+// a hybrid system to the ubuntu data directory. Non-user specific files are
+// copied to <destUbuntuData>/system-data. User specific files are copied to
+// <destUbuntuData>/user-data.
+func copyHybridUbuntuDataAuth(srcUbuntuData, destUbuntuData string) error {
+	destSystemData := filepath.Join(destUbuntuData, "system-data")
+	for _, globEx := range []string{
+		"etc/ssh/*",
+		"etc/sudoers.d/*",
+		"root/.ssh/*",
```

Comment:
I missed seeing a test change in the commit that dropped .profile from here? how are these globs tested?